clar2wasm/src/words/blockinfo.rs

```diff
@@ -278,21 +279,23 @@ mod tests {
     fn get_burn_block_info_non_existent() {
         crosscheck(
             "(get-burn-block-info? header-hash u9999999)",
-            Ok(Some(Value::none())),
+            Ok(Some(
+                Value::some(Value::buff_from([0; 32].to_vec()).unwrap()).unwrap(),
+            )),
         )
     }
 
     #[test]
     fn get_burn_block_info_header_hash() {
         let mut env = TestEnvironment::default();
         env.advance_chain_tip(1);
-        let result = env
-            .evaluate("(get-burn-block-info? header-hash u0)")
-            .expect("Failed to init contract.");
-        assert_eq!(
-            result,
-            Some(Value::some(Value::buff_from([0; 32].to_vec()).unwrap()).unwrap())
-        );
+
+        crosscheck(
+            "(get-burn-block-info? header-hash u0)",
+            Ok(Some(
+                Value::some(Value::buff_from([0; 32].to_vec()).unwrap()).unwrap(),
+            )),
+        )
```

Comment:
`env` is not used anymore. I find it weird that with a default or a modified `env`, the expected result is identical. Is this test really relevant?

Reply:
I don't know if the test is relevant, but I removed the unused env

Reply:
Okay, that seems super weird to me, but it's unrelated to the current issue PR. I guess I'll ask directly to @obycode since he wrote those tests why with or without env the result is identical.

Reply:
I think there are 2 things at play here that we should double check:
1. In epoch 3.0, you can get the burn block info for the current burn block, not just previous burn blocks
2. Since this block info is mocked, the mock implementation seems to support looking up blocks that haven't happened yet. For example, I noticed that the above test `get_burn_block_info_non_existent` was modified to return `(some ...)` instead of `none`. If it's easy enough to adjust our mock implementation so that we get the correct behavior, I think that would be preferred.